Lib/test/test_http_cookiejar.py

```diff
@@ -920,6 +920,13 @@ def test_two_component_domain_ns(self):
 ##         self.assertEqual(len(c), 2)
         self.assertEqual(len(c), 4)
 
+    def test_localhost_domain(self):
+        c = CookieJar()
+
+        interact_netscape(c, "http://localhost", "foo=bar; domain=localhost;")
+
+        self.assertEqual(len(c), 1)
```

Comment:
Can you also test that the cookie in the jar is what you expect it to be?

Reply:
I've written such a test, and it looks like it saves the cookie under ".localhost". Without "domain" attribute present, it saves them under "localhost.local". I'm not so familiar with cookie processing as well, so I'm not really sure if "domain" attribute should affect this in such a way. I've commited the tests in [cec78ed](https://github.com/python/cpython/pull/30108/commits/cec78ed3eae6c648328338aefe4f47c8b59e6c54).

Reply:
Yes, that's a bit odd. It might be fine though as long as it behaves consistently. The public interface doesn't expose the domain names as far as I can see.